openbas-api/src/main/java/io/openbas/fulltextsearch/FullTextSearchService.java

```diff
@@ -0,0 +1,216 @@
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static io.openbas.utils.pagination.PaginationUtils.buildPaginationJPA;
+import static io.openbas.utils.pagination.SortUtilsRuntime.toSortRuntime;
+import static org.springframework.util.StringUtils.hasText;
+
+@Component
+@RequiredArgsConstructor
+public class FullTextSearchService<T extends Base> {
+
+  private final AssetRepository assetRepository;
+  private final AssetGroupRepository assetGroupRepository;
+  private final UserRepository userRepository;
+  private final TeamRepository teamRepository;
+  private final OrganizationRepository organizationRepository;
+  private final ScenarioRepository scenarioRepository;
+  private final ExerciseRepository exerciseRepository;
+
+  private Map<Class<T>, JpaSpecificationExecutor<T>> repositoryMap;
+
+  @PostConstruct
+  @SuppressWarnings("unchecked")
+  public void init() {
+    this.repositoryMap = Map.of(
+        (Class<T>) Asset.class, (JpaSpecificationExecutor<T>) this.assetRepository,
+        (Class<T>) AssetGroup.class, (JpaSpecificationExecutor<T>) this.assetGroupRepository,
+        (Class<T>) User.class, (JpaSpecificationExecutor<T>) this.userRepository,
+        (Class<T>) Team.class, (JpaSpecificationExecutor<T>) this.teamRepository,
+        (Class<T>) Organization.class, (JpaSpecificationExecutor<T>) this.organizationRepository,
+        (Class<T>) Scenario.class, (JpaSpecificationExecutor<T>) this.scenarioRepository,
+        (Class<T>) Exercise.class, (JpaSpecificationExecutor<T>) this.exerciseRepository
+    );
+  }
+
+  public Page<FullTextSearchResult> fullTextSearch(
+      @NotBlank final String clazz,
+      @NotNull final SearchPaginationInput searchPaginationInput) throws ClassNotFoundException {
+    if (!hasText(searchPaginationInput.getTextSearch())) {
+      Pageable pageable = PageRequest.of(searchPaginationInput.getPage(), searchPaginationInput.getSize(), toSortRuntime(searchPaginationInput.getSorts()));
+      return new PageImpl<>(Collections.emptyList(), pageable, 0);
+    }
+
+    Class<?> clazzUnknown = Class.forName(clazz);
+    Class<T> clazzT = this.repositoryMap.keySet().stream()
+        .filter((k) -> k.isAssignableFrom(clazzUnknown))
+        .findFirst()
+        .orElseThrow(() -> new IllegalArgumentException(clazz + " is not handle by full text search"));
+
+    JpaSpecificationExecutor<T> repository = repositoryMap.get(clazzT);
+
+    return buildPaginationJPA(
+        repository::findAll,
+        searchPaginationInput,
+        clazzT
+    ).map(this::transform);
+  }
```

Comment:
If all the results of the transform() are null, we can handle it by returning something like: new PageImpl<>(Collections.emptyList(), pageable, 0) (line 64), in order to send a same type of response ?

Reply:
If the result is null, it will be handle in the buildPaginationJPA method